src/libmain/common-args.cc

```diff
@@ -54,7 +54,9 @@ MixCommonArgs::MixCommonArgs(const string & programName)
         .description = "format of log output; \"raw\", \"internal-json\", \"bar\" "
                         "or \"bar-with-logs\"",
         .labels = {"format"},
-        .handler = {[](std::string format) { setLogFormat(format); }},
+        .handler = {[=](std::string format) {
+            settings.set("log-format", format);
+        }},
```

Comment:
It should be possible to remove the `--log-format` option from `MixCommonArgs`, since any `nix.conf` option is automatically also a command line option.

Reply:
But then it is hidden, and only seeable by --help-config

Reply:
That's a separate issue because it's true for all config options.

Reply:
Not true for max-jobs?

Reply:
I kept it because I specialised Settings and removed the --option option instead